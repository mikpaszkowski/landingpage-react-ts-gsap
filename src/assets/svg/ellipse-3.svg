<svg width="43" height="32" viewBox="0 0 43 32" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M43 21.3145C43 33.0862 33.3741 31.9718 21.5 31.9718C9.62588 31.9718 0 33.0862 0 21.3145C0 9.54283 9.62588 0 21.5 0C33.3741 0 43 9.54283 43 21.3145Z" fill="#FABD20"/>
</svg>
