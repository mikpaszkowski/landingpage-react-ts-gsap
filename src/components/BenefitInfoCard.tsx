import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../styles/styleConstants";
import { device } from "../styles/responsive";

const BenefitInfoCardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-self: center;
`;

const HeadLine = styled.h1`
  font-family: ${fontTypes.primaryFont};
  font-size: 2.4rem;
  padding-top: 8.1rem;
  margin-bottom: 1.6rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: center;
  font-feature-settings: "haln" on;
`;

const Text = styled.p`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  text-align: center;
  margin: 0 7rem;
  margin-bottom: 4.5rem;
`;

const imageCenteringStyle = {
  margin: "0 auto",
  marginTop: "4.5rem",
};

interface BenefitInfoCardProps {
  headline: string;
  description: string;
  image: any;
}

export const BenefitInfoCard = ({
  headline,
  description,
  image,
}: BenefitInfoCardProps) => {
  const BenefitImage = image;
  return (
    <BenefitInfoCardWrapper>
      <BenefitImage style={imageCenteringStyle} />
      <HeadLine>{headline}</HeadLine>
      <Text>{description}</Text>
    </BenefitInfoCardWrapper>
  );
};
