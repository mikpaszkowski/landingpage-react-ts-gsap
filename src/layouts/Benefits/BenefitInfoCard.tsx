import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../styles/constants";
import { device } from "../../styles/responsive";
import { lightTheme } from "../../styles/theme";

const BenefitInfoCardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-self: center;
  height: 45rem;
  border-radius: 8px;
  background-color: ${({ theme }) =>
    theme === lightTheme
      ? theme.palette.common.white
      : theme.palette.common.black};
  transition: all 0.3s cubic-bezier(0.455, 0.03, 0.515, 0.955);
  will-change: transform;

  &:hover {
    box-shadow: 0px 1rem 2rem rgba(0, 0, 0, 0.25);
    transform: translateY(1rem);
  }

  @media ${device.mobileMedium} {
    margin-bottom: 4rem;
  }
`;

const HeadLine = styled.h1`
  font-family: ${fontTypes.primaryFont};
  font-size: 2.4rem;
  padding-top: 8.1rem;
  margin-bottom: 1.6rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: center;
  font-feature-settings: "haln" on;
  color: ${({ theme }) => theme.palette.primary.font};
`;

const Text = styled.p`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  text-align: center;
  margin: 0 7rem;
  margin-bottom: 4.5rem;

  @media ${device.mobileMedium} {
    margin: 0 auto;
    margin-bottom: 4.5rem;
    width: 23rem;
  }
`;

const imageCenteringStyle = {
  margin: "0 auto",
  marginTop: "4.5rem",
};

interface IBenefitInfoCardProps {
  headline: string;
  description: string;
  image: React.FunctionComponent<React.SVGProps<SVGSVGElement>>;
  ref?: React.RefObject<HTMLDivElement>;
  id?: string;
}

export const BenefitInfoCard = React.forwardRef<
  HTMLDivElement,
  IBenefitInfoCardProps
>(({ image, id, headline, description }, ref) => {
  const BenefitImage = image;

  return (
    <BenefitInfoCardWrapper ref={ref} id={id}>
      <BenefitImage style={imageCenteringStyle} />
      <HeadLine>{headline}</HeadLine>
      <Text>{description}</Text>
    </BenefitInfoCardWrapper>
  );
});
