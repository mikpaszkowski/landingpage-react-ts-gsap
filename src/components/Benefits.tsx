import * as React from "react";
import { BenefitInfoCard } from "./BenefitInfoCard";
import styled from "styled-components";
import { ReactComponent as FirstBenefitImage } from "../assets/images/svg-images/benefit-1.svg";
import { ReactComponent as SecondBenefitImage } from "../assets/images/svg-images/benefit-2.svg";
import { ReactComponent as ThirdBenefitImage } from "../assets/images/svg-images/benefit-3.svg";

const BenefitsWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 3rem;
`;

export const Benefits = () => {
  return (
    <BenefitsWrapper>
      <BenefitInfoCard
        headline="1 FREE chat session"
        description="Start with a free text based chat session online"
        image={FirstBenefitImage}
      />
      <BenefitInfoCard
        headline="Personal Coach"
        description="A dedicated coach to help you through your journey"
        image={SecondBenefitImage}
      />
      <BenefitInfoCard
        headline="Text chat session"
        description="An excellent way of preaparing a projects wireframe"
        image={ThirdBenefitImage}
      />
    </BenefitsWrapper>
  );
};
