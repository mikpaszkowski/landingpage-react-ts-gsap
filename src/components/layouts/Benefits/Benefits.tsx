import * as React from "react";
import { BenefitInfoCard } from "./BenefitInfoCard";
import styled from "styled-components";
import { ReactComponent as FirstBenefitImage } from "./benefit-1.svg";
import { ReactComponent as SecondBenefitImage } from "./benefit-2.svg";
import { ReactComponent as ThirdBenefitImage } from "./benefit-3.svg";
import { device } from "../../../styles/responsive";

import { laptopLarge, mobileMedium } from "../../../animations/benefitCards";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
gsap.registerPlugin(ScrollTrigger);

const BenefitsWrapper = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  grid-gap: 3rem;
  opacity: 1;

  @media ${device.mobileMedium} {
    display: block;
    padding: 0 2.4rem;
    margin-top: 3.6rem;
  }
`;

const benefitCardsContent = [
  {
    id: 1,
    headline: "1 FREE chat session",
    description: "Start with a free text based chat session online",
    image: FirstBenefitImage,
  },
  {
    id: 2,
    headline: "Personal Coach",
    description: "A dedicated coach to help you through your journey",
    image: SecondBenefitImage,
  },
  {
    id: 3,
    headline: "Text chat session",
    description: "An excellent way of preaparing a projects wireframe",
    image: ThirdBenefitImage,
  },
];

export const Benefits = () => {
  const { useRef, useEffect } = React;

  const benefitsRef = useRef<HTMLDivElement>(null);
  const benefitsCardsRef = useRef<Array<HTMLDivElement>>([]);
  benefitsCardsRef.current = [];

  const addToRefs = (el: any) => {
    if (el && !benefitsCardsRef.current.includes(el)) {
      benefitsCardsRef.current.push(el);
    }
  };

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": function () {
        benefitsCardsRef.current.forEach((card) => laptopLarge(card));
      },
      "(max-width: 375px)": function () {
        benefitsCardsRef.current.forEach((card) => mobileMedium(card));
      },
    });
  }, []);

  return (
    <BenefitsWrapper ref={benefitsRef}>
      {benefitCardsContent.map(({ id, ...props }) => (
        <BenefitInfoCard
          id={`benefit-card${id}`}
          key={id}
          {...props}
          ref={addToRefs}
        />
      ))}
    </BenefitsWrapper>
  );
};
