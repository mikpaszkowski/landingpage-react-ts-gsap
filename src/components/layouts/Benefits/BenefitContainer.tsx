//Here is the container with the grid contains all of the benefits from the service
import * as React from "react";
import styled from "styled-components";
import { fontTypes } from "../../../styles/constants";
import { Benefits } from "./Benefits";
import { device } from "../../../styles/responsive";
import { benefitCardsContent } from "../../staticContents/benefits/beneifts";
import {
  laptopLarge,
  mobileMedium,
} from "../../../animations/benefitContainer";
import gsap from "gsap";
import { ScrollTrigger } from "gsap/dist/ScrollTrigger";
gsap.registerPlugin(ScrollTrigger);

const HeadLine = styled.h1`
  font-family: ${fontTypes.primaryFont};
  font-size: 8rem;
  padding-top: 6.4rem;
  margin-bottom: 1.6rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: left;
  font-feature-settings: "haln" on;

  @media ${device.mobileLarge} {
    font-size: 5rem;
    margin-bottom: 3.6rem;
  }
`;

const Text = styled.span`
  font-family: ${fontTypes.secondaryFont};
  font-style: normal;
  font-weight: normal;
  font-size: 2.4rem;
  line-height: 3.6rem;
  text-align: left;
  margin-bottom: 7.2rem;
  width: 59.6rem;

  @media ${device.mobileMedium} {
    font-size: 2rem;
    line-height: 3rem;
  }
`;

const BenefitsWrapper = styled.div`
  margin: 0 16.5rem 11.4rem 16.5rem;
  will-change: transform;

  @media ${device.laptopSmall} {
    margin: 0 6rem;
  }

  @media ${device.mobileLarge} {
    margin: 0 2.4rem 6.4rem 2.4rem;
  }
`;

export const BenefitContainer = () => {
  const { useRef, useEffect } = React;

  const sectionRef = useRef<HTMLDivElement>(null);

  useEffect(() => {
    ScrollTrigger.matchMedia({
      "(min-width: 1440px)": () => laptopLarge(sectionRef),
      "(min-width: 500px)": () => laptopLarge(sectionRef),
      "(max-width: 499px)": () => mobileMedium(sectionRef),
    });
  });

  return (
    <BenefitsWrapper ref={sectionRef}>
      <HeadLine>What youâ€™re getting?</HeadLine>
      <Text>
        We bring you personalized development programs that are backed by
        research and crafted from the years of experience
      </Text>
      <Benefits benefitCardsContent={benefitCardsContent} />
    </BenefitsWrapper>
  );
};
