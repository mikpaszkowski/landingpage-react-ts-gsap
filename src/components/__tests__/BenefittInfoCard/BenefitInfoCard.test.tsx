import { cleanup } from "@testing-library/react";
import "jest-styled-components";
import { renderWithDarkTheme, mountWithLightTheme } from "../helpers/index";
import { BenefitInfoCard } from "../../layouts/Benefits/BenefitInfoCard";
import { ReactComponent as FirstBenefitImage } from "../../layouts/Benefits/benefit-1.svg";

afterEach(() => {
  cleanup();
});

window.matchMedia =
  window.matchMedia ||
  function () {
    return {
      matches: false,
      addListener: function () {},
      removeListener: function () {},
    };
  };

it("check render correctly <BenefitInfoCard /> component - darkTheme", () => {
  const benefitCardsContent = {
    id: "1",
    headline: "1 FREE chat session",
    description: "Start with a free text based chat session online",
    image: FirstBenefitImage,
    ref: null,
  };

  const tree = renderWithDarkTheme(
    <BenefitInfoCard {...benefitCardsContent} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

// describe("check BenfitInfoCard css properties", () => {
//   const benefitCardsContent = {
//     id: "1",
//     headline: "1 FREE chat session",
//     description: "Start with a free text based chat session online",
//     image: FirstBenefitImage,
//     ref: null,
//   };

//   const wrapper = mountWithLightTheme(
//     <BenefitInfoCard {...benefitCardsContent} />
//   ).find(BenefitInfoCard);

//   it("should have #FFFFFF background-color - lightTheme", () => {
//     expect(wrapper).toHaveStyleRule("background-color", "#FFFFFF");
//   });
// });
