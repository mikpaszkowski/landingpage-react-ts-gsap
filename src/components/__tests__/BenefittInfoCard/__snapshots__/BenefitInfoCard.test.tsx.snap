// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check render correctly <BenefitInfoCard /> component - darkTheme 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  height: 45rem;
  border-radius: 8px;
  background-color: #131E07;
  -webkit-transition: all 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  transition: all 0.3s cubic-bezier(0.455,0.03,0.515,0.955);
  will-change: transform;
}

.c0:hover {
  box-shadow: 0px 1rem 2rem rgba(0,0,0,0.25);
  -webkit-transform: translateY(1rem);
  -ms-transform: translateY(1rem);
  transform: translateY(1rem);
}

.c1 {
  font-family: 'Rozha One',serif;
  font-size: 2.4rem;
  padding-top: 8.1rem;
  margin-bottom: 1.6rem;
  font-style: normal;
  font-weight: normal;
  line-height: 117%;
  text-align: center;
  font-feature-settings: "haln" on;
  color: #FFFFFF;
}

.c2 {
  font-family: 'Reem Kufi',sans-serif;
  font-style: normal;
  font-weight: normal;
  font-size: 1.8rem;
  line-height: 2.7rem;
  text-align: center;
  margin: 0 7rem;
  margin-bottom: 4.5rem;
}

@media (max-width:375px) {
  .c2 {
    margin: 0 auto;
    margin-bottom: 4.5rem;
    width: 23rem;
  }
}

<div
  className="c0"
  id="1"
>
  <svg
    style={
      Object {
        "margin": "0 auto",
        "marginTop": "4.5rem",
      }
    }
  >
    benefit-1.svg
  </svg>
  <h1
    className="c1"
  >
    1 FREE chat session
  </h1>
  <p
    className="c2"
  >
    Start with a free text based chat session online
  </p>
</div>
`;
